ddtrace/context.py

```diff
@@ -123,11 +130,81 @@ def sampling_priority(self, value):
     @property
     def _traceparent(self):
         # type: () -> str
-        if self.trace_id is None or self.span_id is None:
+        tp = self._meta.get(_TRACEPARENT_KEY)
+        dd_trace_id = self.trace_id
+        span_id = self.span_id
+
+        # if we only have a traceparent then we'll forward it
+        if tp and (span_id is None or dd_trace_id is None):
+            return tp
+
+        # determine the trace_id value
+        if tp:
+            # grab the original traceparent trace id, not the converted value
+            trace_id = tp.split("-")[1]
+        elif dd_trace_id:
+            trace_id = "{:032x}".format(dd_trace_id)
+        else:
+            # if we don't have trace id value we can't build a traceparent
+            trace_id = ""
+
+        # if we don't have trace id value or a span id value we can't build a traceparent
+        if not trace_id or not span_id:
```

Comment:
if we make the changes above then we don't need to check trace_id again. 

```suggestion
        # if span_id is not set then a valid traceparent can not be set
        if not span_id:
```